[leptonica] pkgconfig --exists lept fails

### Description / Steps to reproduce the issue

During the development of a small app with Qt6 and Tesseract OCR I ran into CMake configuration problem.

Even though all needed packages are installed CMake is not able to locate the Tesseract package due to its dependency Leptonica not being able to find some of its dependencies (libpng, libjpeg, libtiff).

```
-- Building for: Ninja
-- The CXX compiler identification is GNU 15.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: X:/msys64/mingw64/bin/c++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The C compiler identification is GNU 15.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: X:/msys64/mingw64/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found WrapVulkanHeaders: X:/msys64/mingw64/include
-- Found PkgConfig: X:/msys64/mingw64/bin/pkg-config.exe (found version "2.5.1")
-- Checking for module 'tesseract'
--   Package 'PNG', required by 'lept', not found
Package 'JPEG', required by 'lept', not found
Package 'TIFF', required by 'lept', not found
CMake Error at X:/msys64/mingw64/share/cmake/Modules/FindPkgConfig.cmake:639 (message):
  The following required packages were not found:

   - tesseract

Call Stack (most recent call first):
  X:/msys64/mingw64/share/cmake/Modules/FindPkgConfig.cmake:867 (_pkg_check_modules_internal)
  CMakeLists.txt:22 (pkg_check_modules)
```

The supplied lept.pc of the leptonica package contains the following:
```
prefix=/mingw64
exec_prefix=${prefix}/bin
libdir=${prefix}/lib
includedir=${prefix}/include

Name: leptonica
Description: An open source C library for efficient image processing and image analysis operations
Version: 1.86.0
Requires.private:  ZLIB PNG JPEG TIFF libwebp libwebpmux libopenjp2
Libs: -L${libdir} -lleptonica
Libs.private:  -lgif -lm
Cflags: -I${includedir} -I${includedir}/leptonica
```

Its `Requires.private` tries to locate `png.pc` `jpeg.pc` and `tiff.pc` but these dependencies installed via pacman for leptonica supply pkgconfig files with the names `libpng.pc`, `libjpeg.pc` and `libtiff-4.pc`.

Changing the `Requires.private:` by replacing the 3 dependencies with their proper pkgconfig file names fixes the problem.

To reproduce the problem you can follow the MSYS2 build steps for my project at:
https://github.com/smokejohn/deadclock/tree/master/?tab=readme-ov-file#building-on-windows

### Expected behavior

CMake should be able to find the tesseract package via pkgconfig.

### Actual behavior

CMake is unable to locate the tesseract package due to leptonica not finding all its dependencies leading to `pkgconf --exists tesseract`  and `pkgconf --exists lept` to fail and return 1.

### Verification

- [x] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-26100

### MINGW environments affected

- [x] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANGARM64

### Are you willing to submit a PR?

Yes

--- a/mingw-w64-leptonica/PKGBUILD
+++ b/mingw-w64-leptonica/PKGBUILD
@@ -4,7 +4,7 @@ _realname=leptonica
 pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
 pkgver=1.86.0
-pkgrel=2
+pkgrel=3
 pkgdesc="An open source C library for efficient image processing and image analysis operations (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
@@ -30,11 +30,13 @@ checkdepends=("${MINGW_PACKAGE_PREFIX}-gnuplot")
 source=("https://github.com/DanBloomberg/leptonica/archive/${pkgver}/${_realname}-${pkgver}.tar.gz"
         "001-Fix-SOVERSION-on-mingw.patch"
         "002-Fix-pkgconfig-file-name.patch"
-        "003-Fix-cmake-include-directories.patch")
+        "003-Fix-cmake-include-directories.patch"
+        "https://github.com/DanBloomberg/leptonica/pull/778.patch")
 sha256sums=('b4447faf61a8786a9b211d58d4103d85d47fd3a5dd418d5a6bc525d41db54ccc'
             '0bba2b4773083b09266b94e759801b07e787ef8bdc3458881202f06c23f03e8b'
             '6a9bde25199eba47e12d0013591fb4d376df386b1dccd252a8b64374e742f889'
-            'd278ea2957025c9cc9bccbdbf1c757e1ce2f29d14287b0dfbcac02ebebc48f88')
+            'd278ea2957025c9cc9bccbdbf1c757e1ce2f29d14287b0dfbcac02ebebc48f88'
+            'f411e8f7a7bb7d4996c7e1e7e0f7ca49aae189e1269efd17b466fe9c069e020d')
 
 apply_patch_with_msg() {
   for _patch in "$@"
@@ -49,7 +51,8 @@ prepare() {
   apply_patch_with_msg \
     001-Fix-SOVERSION-on-mingw.patch \
     002-Fix-pkgconfig-file-name.patch \
-    003-Fix-cmake-include-directories.patch
+    003-Fix-cmake-include-directories.patch \
+    778.patch
 }
 
 build() {

